src/main/kotlin/io/sledge/deployer/crx/CrxDeployer.kt

```diff
@@ -1,33 +1,31 @@
 package io.sledge.deployer.crx
 
 import com.github.ajalt.clikt.output.TermUi.echo
-import io.sledge.deployer.core.api.Deployer
-import io.sledge.deployer.core.api.Deployment
-import khttp.extensions.fileLike
-import khttp.post
+import io.sledge.deployer.crx.command.*
+import io.sledge.deployer.http.HttpClient
+import io.sledge.deployer.crx.zip.Unarchiver
+import io.sledge.deployer.crx.zip.Unarchiver.Companion.VLT_PROPERTIES
 import java.io.File
 
-class CrxDeployer : Deployer {
+class CrxDeployer {
 
-    override fun deploy(deployment: Deployment) {
-        echo("Start deployment for ${deployment.deploymentDef.name}")
-        for (artifact in deployment.deploymentDef.deploymentArtifacts) {
-            echo("Deploying artifact: ${artifact.filePath}...")
+    fun deploy(configuration: CrxConfiguration) {
 
-            // Upload: curl -u admin:admin -F cmd=upload -F force=true -F package=@test.zip http://localhost:4502/crx/packmgr/service/.json
-
-            val url = "http://httpbin.org/post"
-            val files = listOf(File(artifact.filePath).fileLike())
-
-            val r = post(url, files = files)
-
-            echo("${r.text}")
-
-            // Install: curl -u admin:admin -F cmd=install http://localhost:4502/crx/packmgr/service/.json/etc/packages/my_packages/test.zip
-
-            // Uninstall: curl -u admin:admin -F cmd=uninstall http://localhost:4502/crx/packmgr/service/.json/etc/packages/my_packages/test.zip
+        if (configuration == null) {
+            echo("ERROR: Deployment definition name not found in Sledge file.", err = true)
+        }
 
-            // Delete: curl -u admin:admin -F cmd=delete http://localhost:4502/crx/packmgr/service/.json/etc/packages/my_packages/test.zip
+        val httpClient = HttpClient(configuration)
+        configuration.deploymentDefinition.let {
+            echo("Start deployment for ${configuration.deploymentDefinition.name}")
+            for (artifact in configuration.deploymentDefinition.deploymentArtifacts) {
+                val packageName = Unarchiver().unzipPropertiesFile(artifact.filePath, VLT_PROPERTIES)
```

Comment:
It is a bit misleading, we set the packageName, but the method is called unzipPropertiesFile....maybe we should rename it to "extractPackageName" or so.